Assets/Scripts/SceneState/DclTransformComponent.cs

```diff
@@ -20,67 +20,112 @@ public DclTransformComponent(DclComponent c) : base(c.NameInCode, c.NameInCode)
 
         public DclComponentProperty<Vector3> Scale => GetPropertyByName("scale")?.GetConcrete<Vector3>();
 
-        // gives the global fixed position
-        public Vector3 GlobalFixedPosition
+        public Matrix4x4 GlobalTransformMatrix
         {
             get
             {
                 if (Entity.Parent == null)
                 {
-                    return Position.FixedValue;
+                    Matrix4x4 transformMatrix = new Matrix4x4();
+                    transformMatrix.SetTRS(Position.Value, Rotation.Value, Scale.Value);
+                    return transformMatrix;
                 }
-
+                
                 var parentTransform = Entity.Parent.GetTransformComponent();
-                var scaledLocalPosition = Position.FixedValue;
-                scaledLocalPosition.Scale(parentTransform.Scale.FixedValue);
-                var globalPosition = (parentTransform.GlobalFixedPosition + (parentTransform.GlobalFixedRotation * scaledLocalPosition));
+                var parentMatrix = parentTransform.GlobalTransformMatrix;
+
+                Matrix4x4 localMatrix = new Matrix4x4();
+                localMatrix.SetTRS(Position.Value, Rotation.Value, Scale.Value);
+
+                var globalMatrix = parentMatrix * localMatrix;
 
-                return globalPosition;
+                return globalMatrix;
+            }
+        }
+
+        public Quaternion GlobalRotation
```

Comment:
Please move the GlobalRotation under the GlobalPosition, so we have consistent sorting